block/src/Pos/Block/Worker.hs

```diff
@@ -107,9 +112,20 @@ blkCreatorWorker :: BlockWorkMode ctx m => (WorkerSpec m, OutSpecs)
 blkCreatorWorker =
     onNewSlotWorker onsp mempty $ \slotId diffusion ->
         recoveryCommGuard "onNewSlot worker, blkCreatorWorker" $
-        blockCreator slotId diffusion `catchAny` onBlockCreatorException
+        catchReportExc $
+        blockCreator slotId diffusion
   where
-    onBlockCreatorException = reportOrLogE "blockCreator failed: "
+    -- Detect errors that should not crash the worker (and the node with it).
+    isNonCriticalExc = \case
+        -- TODO: [CSL-198] Revise this list of errors.
```

Comment:
Are you going to do it in near time? I mean, I except CSL-198 to be closed in next few days, and if it's closed, there shouldn't be such comment.
I would test scenario when network connection disappears (dunno how to do it on localhost, but I guess there is a way) to know which exception is thrown in this case (and not sure if it will be `DialogUnexpected`) and I think it will be enough.